src/meta/meta_bulk_load_service.h

```diff
@@ -290,6 +290,18 @@ class bulk_load_service
     ///
     /// helper functions
     ///
+    inline std::shared_ptr<app_state> get_app(const std::string &name)
+    {
+        zauto_read_lock l(app_lock());
+        return _state->get_app(name);
+    }
+
+    inline std::shared_ptr<app_state> get_app(int32_t app_id)
+    {
+        zauto_read_lock l(app_lock());
+        return _state->get_app(app_id);
```

Comment:
The lifecycle is maintained by _state, so I think there is no need to return std::shared_ptr here